src/runtime/policy/policy.ts

```diff
@@ -279,6 +290,60 @@ export class PolicyField {
       label: label === 'raw' ? '' : label,
     };
   }
+
+  // Returns policy fields in the format of Schema fields.
+  toSchemaFields(parentType: Type) {
+    const field = parentType.getEntitySchema().fields[this.name];
+    return {name: this.name, field: this.toSchemaField(field)};
+  }
+
+  private toSchemaField(field) {
+    switch (field.kind) {
+      case 'kotlin-primitive':
+      case 'schema-primitive': {
+        assert(this.subfields.length === 0);
+        return field;
+      }
+      case 'schema-collection':
+      case 'schema-ordered-list':
+      case 'schema-nested': {
+        return FieldType.create(
+          {kind: field.kind, schema: this.toSchemaField(field.schema)}
+        );
+      }
+      case 'schema-reference': {
```

Comment:
optional - schema-field.ts has `Kind` enum with these values, might be nicer to use them.